[WPE] WPE Platform: make functions to get DRM device and render node members of WPEDisplay

carlosgcampos opened the pull request:
#### c2361c31a74159368c4298faa9049cd0cc3a3453
<pre>
[WPE] WPE Platform: make functions to get DRM device and render node members of WPEDisplay
<a href="https://bugs.webkit.org/show_bug.cgi?id=270629">https://bugs.webkit.org/show_bug.cgi?id=270629</a>

Reviewed by Alejandro G. Castro.

They are currently global, and in the case of DRM might not be
consistent with what it&apos;s actually used. Now that we have the primary
display, we can move the API to WPEDisplay and return what every
platform is actually using.

* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
(wpeBufferDMABufDispose):
(wpeBufferDMABufTryEnsureGBMDevice):
(wpeBufferDMABufImportToPixels):
(gbmDevice): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpe_display_get_drm_device):
(wpe_display_get_drm_render_node):
(renderNodeFile): Deleted.
(wpeDRMRenderNodeFile): Deleted.
(wpe_render_node_device): Deleted.
(renderDeviceFile): Deleted.
(wpeDRMRenderDeviceFile): Deleted.
(wpe_render_device): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/drm/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/drm/RefPtrUdev.cpp: Added.
(WTF::udev&gt;::refIfNotNull):
(WTF::udev&gt;::derefIfNotNull):
(WTF::udev_device&gt;::refIfNotNull):
(WTF::udev_device&gt;::derefIfNotNull):
(WTF::udev_enumerate&gt;::refIfNotNull):
(WTF::udev_enumerate&gt;::derefIfNotNull):
* Source/WebKit/WPEPlatform/wpe/drm/RefPtrUdev.h: Added.
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.cpp:
(WPE::DRM::Seat::create):
(WPE::DRM::Seat::Seat):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(findDevice):
(wpeDisplayDRMConnect):
(wpeDisplayDRMGetDRMDevice):
(wpeDisplayDRMGetDRMRenderNode):
(wpe_display_drm_class_init):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandDispose):
(wpeDisplayWaylandConnect):
(wpeDisplayWaylandGetDRMDevice):
(wpeDisplayWaylandGetDRMRenderNode):
(wpe_display_wayland_class_init):

Canonical link: <a href="https://commits.webkit.org/275830@main">https://commits.webkit.org/275830@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d061331932a9b0e9fcde1b3c21e3ff755257dfe

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42835 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21856 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45236 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45451 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38960 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25541 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19236 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35444 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43408 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18875 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36872 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16412 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16488 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/891 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39005 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38269 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46966 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17662 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14557 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42183 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19287 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40823 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19451 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5820 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18917 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro commented on Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
```diff
@@ -150,45 +157,51 @@ static gpointer wpeBufferDMABufImportToEGLImage(WPEBuffer* buffer, GError** erro
-        close(fd);
-    });
-    return device;
+    auto* priv = buffer->priv;
```
Is it possible that buffer or priv are NULL? Maybe we can add an assertion to check it?

carlosgcampos replied:
No, this is is called from a public method that already checks buffer is a `WPE_BUFFER_DMA_BUF`

alexgcastro reviewed (approved):
LGTM! Great improvement in the rendering device detection.

carlosgcampos committed:
[WPE] WPE Platform: make functions to get DRM device and render node members of WPEDisplay
https://bugs.webkit.org/show_bug.cgi?id=270629

Reviewed by Alejandro G. Castro.

They are currently global, and in the case of DRM might not be
consistent with what it's actually used. Now that we have the primary
display, we can move the API to WPEDisplay and return what every
platform is actually using.

* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
(wpeBufferDMABufDispose):
(wpeBufferDMABufTryEnsureGBMDevice):
(wpeBufferDMABufImportToPixels):
(gbmDevice): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpe_display_get_drm_device):
(wpe_display_get_drm_render_node):
(renderNodeFile): Deleted.
(wpeDRMRenderNodeFile): Deleted.
(wpe_render_node_device): Deleted.
(renderDeviceFile): Deleted.
(wpeDRMRenderDeviceFile): Deleted.
(wpe_render_device): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/drm/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/drm/RefPtrUdev.cpp: Added.
(WTF::udev>::refIfNotNull):
(WTF::udev>::derefIfNotNull):
(WTF::udev_device>::refIfNotNull):
(WTF::udev_device>::derefIfNotNull):
(WTF::udev_enumerate>::refIfNotNull):
(WTF::udev_enumerate>::derefIfNotNull):
* Source/WebKit/WPEPlatform/wpe/drm/RefPtrUdev.h: Added.
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.cpp:
(WPE::DRM::Seat::create):
(WPE::DRM::Seat::Seat):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSeat.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(findDevice):
(wpeDisplayDRMConnect):
(wpeDisplayDRMGetDRMDevice):
(wpeDisplayDRMGetDRMRenderNode):
(wpe_display_drm_class_init):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandDispose):
(wpeDisplayWaylandConnect):
(wpeDisplayWaylandGetDRMDevice):
(wpeDisplayWaylandGetDRMRenderNode):
(wpe_display_wayland_class_init):

Canonical link: https://commits.webkit.org/275830@main
